### ISSUE TITLE
Trimming on instantiation

### ISSUE THREAD

**Issue by rwitchell** [thephpleague/csv#132]
Hi,
I have a concern with the trim on instantiation.

```php
 public static function createFromString($str, $newline = "\n")
    {
        $file = new SplTempFileObject();
        $file->fwrite(rtrim($str).$newline);
```

We allow persons to copy/paste data from excel into a text input, which we then create a CSV object. 

For example, if they copy in 

![image](https://cloud.githubusercontent.com/assets/5378642/10782470/c02509e4-7da4-11e5-9437-a3945784dda4.png)

the rtrim() on createFromString() would remove the final tab character, effectively removing the empty column in the last row.

I don't believe the trim here is a good idea.

The work arounds are:

1. applying the second parameter, $newline, to be "\t\n" - however if data is provided with the last cell filled, then i've needlessly added an extra column. Solution: i have to read the last character of $str to check if it is a tab, and if so, update the $newline to "\t\n" otherwise leave it as "\n"

```php
$newline = substr($_POST['textinput'], -1) === "\t" ? "\t\n" : "\n";
$reader = Reader::createFromString($_POST['textinput'], $newline);
$reader->setDelimiter("\t")
            ->setEnclosure("\"")
            ->setEscape("\\");
```

I don't believe the rtrim is necessary, as the application should match the user's representation of data as closely as possible. I think the better option would be to add trim() as an option on the object, $reader->rtrimIncomingString(true)

thoughts?

**Comment by nyamsprod** [comment]
Hi @rwitchell,

Thanks for reporting this bug. As a matter of fact, this one is being fixed as part of the 7.2.0 release that will be out hopefully this week if I still have time or next week. The change will be in the master branch anyway.

The `$newline` parameter is being deprecated from the `createFromString` method and will be remove completely in the next major release.

**Comment by nyamsprod** [comment]
This issue is fixed in the master branch
